source/jormungandr/jormungandr/scenarios/ridesharing/karos.py

```diff
@@ -111,7 +111,21 @@ def _make_response(self, raw_json):
             res.ridesharing_ad = offer.get('webUrl')
             res.duration = offer.get('duration')
             res.origin_pickup_duration = offer.get('departureToPickupWalkingTime')
+            res.origin_pickup_distance = int(offer.get('departureToPickupWalkingDistance'))
+            res.origin_pickup_shape = []
+            origin_pickup_shape = decode_polyline(offer.get('departureToPickupWalkingPolyline'), precision=5)
+            if origin_pickup_shape:
+                res.origin_pickup_shape.extend(
+                    (type_pb2.GeographicalCoord(lon=c[0], lat=c[1]) for c in origin_pickup_shape)
+                )
             res.dropoff_dest_duration = offer.get('dropoffToArrivalWalkingTime')
+            res.dropoff_dest_distance = int(offer.get('dropoffToArrivalWalkingDistance'))
```

Comment:
As we can not guarantee the type and format of attributes values in any RS connector, isn't it better to transform in ridesharing_service_manager ? 
For example offer.get('dropoffToArrivalWalkingDistance') is float in karos should be transformed to int() as you did above.
I think we should transform value of mostly every attribute of ridesharing_journey (ridesharing_journey.py)

Reply:
:+1: 
it is a good idea to cast with the good format within ridesharing_service_manager